source/sigv4.c

```diff
@@ -1243,57 +1237,59 @@ static SigV4Status_t generateCredentialScope( const SigV4Parameters_t * pSigV4Pa
         assert( canonicalURILen != NULL );
         assert( *canonicalURILen > 0U );
 
-        pUriLoc = pUri;
-        pBufLoc = pCanonicalURI;
         bufferLen = *canonicalURILen;
 
-        while( ( i++ < uriLen ) && ( returnStatus == SigV4Success ) )
+        for( ; ( uriIndex < uriLen ) && ( returnStatus == SigV4Success ); uriIndex++ )
```

Comment:
This is a MISRA 14.2 violations as a `for` loop since `returnStatus` is altered in the loop
```suggestion
        while( ( uriIndex++ < uriLen ) && ( returnStatus == SigV4Success ) )
```